pkg/gui/custom_commands.go

```diff
@@ -118,16 +118,22 @@ func (gui *Gui) menuPrompt(prompt config.CustomCommandPrompt, promptResponses []
 	return gui.createMenu(title, menuItems, createMenuOptions{showCancel: true})
 }
 
-func (gui *Gui) GenerateMenuCandidates(commandOutput string, filter string, format string) ([]string, error) {
+func (gui *Gui) GenerateMenuCandidates(commandOutput string, filter string, tFormat string, dFormat string) ([]string, []string, error) {
 	candidates := []string{}
+	descriptions := []string{}
```

Comment:
Having every menu entry represented in 2 arrays seems wrong to me.  

What about this function returning `[]*menuItem`?  
Maybe you'll need to add a callback as function argument to this function that itself has one argument of the selected value.

Or we have a struct with the label and value inside of it
```
type commandMenuEntry struct {
  label string
  value string
}
```
And return that
```
entries := []commandMenuEntry{}
```

Reply:
I'm not 100% sure I get what you mean, but I agree that's it's better to represent a menu entry with a single entity.
 Now I'm using `labelFormat` to format the label (Goes into description of entries) and `valueFormat` to format the entry's value itself. I hope this is a little clearer.
The end goal was, instead of having a menu like this (searching for commits which mentioned "lazygit" in go files):
![2021-08-06_11-01](https://user-images.githubusercontent.com/34474472/128494106-a6f97d1a-9bb5-4236-a526-d2670250cac8.png)

we can get much more helpful information:
![2021-08-06_11-01_1](https://user-images.githubusercontent.com/34474472/128494168-978125a6-2016-4eb4-8530-cba7471f2d10.png)